### PR TITLE
Clustering/kmeans

### PR DESCRIPTION
**Pull request by AHReccese**
#### Reference Issues/PRs
#17 
#### What does this implement/fix? Explain your changes.
Add Clustering Chain
Add Kmeans(+test) support
Add Clustering purposed dataset and test
Add Clustering model's tests runner
Update `test_pymilo.py` and decouple the prediction from the pymiloing process(enhance the cohesiveness of pymiloing)

### CODE REVIEW

**Comment by AHReccese** [comment]
@sepandhaghighi
Hi, dear Sepand, I hope you're doing well, I suggest you review this major PR til 10/05/2023.
Although it is a major PR, it isn't complex that much.
I can explain the details whenever you want in our morning office times.

File: pymilo/chains/clustering_chain.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+"""PyMilo chain for decision trees."""
```
Comment (sepandhaghighi) [inline pymilo/chains/clustering_chain.py:2]: decision trees?

File: SUPPORTED_MODELS.md [hunk 2]
```diff
@@ -247,3 +247,11 @@
 	</tr>
 </table>
 
+## Clustering Models
+<table>
+	<tr align="center">
+		<th>Library</th>
+		<th>Model Name</th>
+        <th>Category</th>
```
Comment (sepandhaghighi) [inline SUPPORTED_MODELS.md:255]: Please update this file 💯

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]
@AHReccese Thanks for your efforts 🔥 

1. Take a look at my comments
2. Please update `CHANGELOG.md`

File: pymilo/chains/clustering_chain.py (hunk 1 above)
Comment (AHReccese) [reply pymilo/chains/clustering_chain.py]: done.

File: SUPPORTED_MODELS.md (hunk 2 above)
Comment (AHReccese) [reply SUPPORTED_MODELS.md:255]: done.

**Comment by AHReccese** [comment]
@sepandhaghighi 
a kind reminder(since it is 4 days overdue(10 + 4 days), I know we're busy focusing on the proposal, but consider this reminder just as a commitment to the OSL guidelines.

File: pymilo/utils/test_pymilo.py [hunk 3]
```diff
@@ -49,8 +52,6 @@ def pymilo_test(model, model_name, test_data):
     :type test_data: np.ndarray or list
```
Comment (sepandhaghighi) [inline pymilo/utils/test_pymilo.py:52]: remove `test_data` from docstring

File: pymilo/utils/test_pymilo.py [hunk 4]
```diff
@@ -121,7 +122,31 @@ def pymilo_classification_test(classifier, model_name, test_data):
     report_status(comparison_result, model_name)
     return comparison_result
 
+def pymilo_clustering_test(clusterer, model_name, x_test, support_prediction = False):
+    """
+    Test the package's main structure in clustering task.
 
+    :param clusterer: the given clusterer model
```
Comment (sepandhaghighi) [inline pymilo/utils/test_pymilo.py:129]: wrong `docstring`

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]
@AHReccese Sorry for my late response 🔥 
Please take a look at my comments. Additionally, kindly update the `CHANGELOG.md` file. Thank you.

**Review by sepandhaghighi — APPROVED** [review]
🚀

**Pull request merged**